File: 1227-number-of-equivalent-domino-pairs/number-of-equivalent-domino-pairs.java
class Solution {
    public int numEquivDominoPairs(int[][] dominoes) {
        // count = [0,0,...,0]
        int[] count = new int[100]; // max key [9,9] = 99
        int result = 0;

        for (int[] domino : dominoes) {
            int i = domino[0], j = domino[1];
            // [1,2] = [2,1] = 12
            int key = i < j ? i * 10 + j : j * 10 + i;
            result += count[key];
            count[key]++;
        }

        return result;
    }
}

File: 1227-number-of-equivalent-domino-pairs/number-of-equivalent-domino-pairs.go
func numEquivDominoPairs(dominoes [][]int) int {
    count := [100]int{};
    result := 0;

    for _, domino := range dominoes {
        a, b := domino[0], domino[1];
        key := a*10 + b;

        if a > b {
            key = b*10 + a;
        }

        result += count[key];
        count[key]++;
    }

    return result;
}